src/test/java/org/dependencytrack/resources/v1/FindingResourceTest.java

```diff
@@ -589,6 +593,45 @@ public void getAllFindingsGroupedByVulnerabilityWithAclEnabled() {
         Assert.assertEquals(1, json.getJsonObject(2).getJsonObject("vulnerability").getInt("affectedProjectCount"));
     }
 
+    @Test
+    public void getSARIFFindingsByProjectTest() {
+        Project p1 = qm.createProject("Acme Example", null, "1.0", null, null, null, true, false);
+        Component c1 = createComponent(p1, "Component A", "1.0");
+        Component c2 = createComponent(p1, "Component B", "1.0");
+        Component c3 = createComponent(p1, "Component C", "1.0");
+        Vulnerability v1 = createVulnerability("Vuln-1", Severity.CRITICAL);
+        Vulnerability v2 = createVulnerability("Vuln-2", Severity.HIGH);
+        Vulnerability v3 = createVulnerability("Vuln-3", Severity.MEDIUM);
+        Vulnerability v4 = createVulnerability("Vuln-4", Severity.LOW);
+        qm.addVulnerability(v1, c1, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v2, c1, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v3, c2, AnalyzerIdentity.NONE);
+        qm.addVulnerability(v4, c3, AnalyzerIdentity.NONE);
+
+        Response response = target(V1_FINDING + "/project/" + p1.getUuid().toString()).request()
+            .header(HttpHeaders.ACCEPT, MEDIA_TYPE_SARIF_JSON)
+            .header(X_API_KEY, apiKey)
+            .get(Response.class);
+
+        Assert.assertEquals(200, response.getStatus(), 0);
+        Assert.assertEquals(MEDIA_TYPE_SARIF_JSON, response.getHeaderString(HttpHeaders.CONTENT_TYPE));
+
+        JsonObject json = parseJsonObject(response);
+        Assert.assertNotNull(json);
+
+        JsonArray runs = json.getJsonArray("runs");
+        Assert.assertNotNull(runs);
+        JsonObject runsJson = runs.getJsonObject(0);
+        Assert.assertNotNull(runsJson);
+        Assert.assertEquals("OWASP Dependency-Track", runsJson.getJsonObject("tool").getJsonObject("driver").getString("name"));
+        Assert.assertEquals(new About().getVersion(), runsJson.getJsonObject("tool").getJsonObject("driver").getString("version"));
+        Assert.assertNotNull(runsJson.getJsonObject("tool").getJsonObject("driver").getJsonArray("rules"));
+        Assert.assertEquals("Vuln-1", runsJson.getJsonObject("tool").getJsonObject("driver").getJsonArray("rules").getJsonObject(0).getString("id"));
+        Assert.assertNotNull(runsJson.getJsonArray("results"));
+        Assert.assertEquals("error", runsJson.getJsonArray("results").getJsonObject(0).getString("level"));
+        Assert.assertEquals("Vuln-1", runsJson.getJsonArray("results").getJsonObject(0).getString("ruleId"));
```

Comment:
It would be great to assert the entire content of the JSON output here. That makes it easier to spot regressions later. You can find an example of how this can be done here:

https://github.com/DependencyTrack/dependency-track/blob/3b52284f98526f533bd8b7b705e707f6f5847102/src/test/java/org/dependencytrack/resources/v1/BomResourceTest.java#L552-L661

Reply:
Thanks a lot @nscuro ! This is definitely a better way. Will update